Added pagination on reporters. Get program memberships/groups types and  users  that manage program

CR1AT0RS opened the pull request:
Added pagination on reporters. Added attributes on reporters to choose page_size or get a page_number. 

example uage:
program.reporters(page_number: 1, page_size: 25)

esjee commented:
Cool! The code looks good to me, but could you add a test?

oreoshake commented:
It would be cool if there were an autopagination option like octokit has (maybe it’s implemented in a gem or something). These seem like good defaults to start with so we can add autopagination later.

CR1AT0RS commented:
Added: Get all users that manage the program. Program groups and membership type. Updated documentation. For auto pagination people can just a add for loop in their code.

oreoshake commented on lib/hackerone/client/program.rb:
```diff
@@ -24,6 +24,12 @@ def id
         OpenStruct.new(@program[:attributes])
       end
+       # List all groups, membership types and users for a program
+      def users()
```
I wonder if it's better to just expose the private `members` method instead.
